spec/types.md

```diff
@@ -2,385 +2,418 @@
 
 ## General
```

Comment:
This file was already covered by the previous PR. Are there new changes?

Reply:
Sorry @BillWagner no new changes here. I'll refresh my fork moving forward. Thanks!